neurodamus/io/synapse_reader.py

```diff
@@ -253,27 +255,57 @@ def get_property(self, gid, field_name):
         """
         return self._data[gid][field_name]
 
-    def preload_data(self, ids):
-        """Preload SONATA fields for the specified IDs"""
+    def preload_data(self, gids, minimal_mode=False):
+        """
+        Preload SONATA fields for the specified IDs.
+        Set minimal_mode to True to read a single synapse per connection
+        """
+        # Ensure we dont ask synapses for more than 50 target cells at once
+        CHUNK_SIZE = 50
+        if len(gids) <= CHUNK_SIZE:
+            return self._preload_data_chunk(gids, minimal_mode)
+
+        ranges = list(gen_ranges(len(gids), CHUNK_SIZE))
+        for start, end in ProgressBar.iter(ranges, name="Prefetching"):
+            self._preload_data_chunk(gids[start:end], minimal_mode)
```

Comment:
yes. We never really had to care for extreme cases, but chunking is important to avoid exploding memory in case we have many cells in a single machine.

Reply:
Hm, I think we originally try to read synapses completely to minimize the amount of separate accesses to GPFS. Can we preserve that behavior?

Reply:
We can adjust the CHUNK_SIZE to get as much data as possible. But I would like to cap memory consumption. With a CHUNK of 1000 cells, with 1000 syns/cell that's up to 1m rows per property at a time. Would it sounds reasonable?

Reply:
Never mind, even if chunking is interesting we are doing collective IO, so the operation is not as trivial.
I added a comment and activated chunking only in crash-test